edgraph/server.go

```diff
@@ -1544,8 +1544,30 @@ func parseRequest(qc *queryContext) error {
 			if err != nil {
 				return err
 			}
+			newGmu := dql.Mutation{Set: make([]*api.NQuad, 0)}
+			if isUnique(gmu, ctx) {
+				for _, singlePred := range gmu.Set {
+
+					predSchema, _ := schema.State().Get(ctx, fmt.Sprint(singlePred.GetNamespace())+"-"+singlePred.GetPredicate())
+					if predSchema.Unique {
```

Comment:
we already checked in isUnique whether the mutation has a predicate with unique schema. Why are we checking here again?